benches/flamegraph_profiler.rs

```diff
@@ -0,0 +1,38 @@
+use std::{fs::File, os::raw::c_int, path::Path};
+
+use criterion::profiler::Profiler;
+use pprof::ProfilerGuard;
+
+pub struct FlamegraphProfiler<'a> {
+    frequency: c_int,
+    active_profiler: Option<ProfilerGuard<'a>>,
+}
+
+impl<'a> FlamegraphProfiler<'a> {
+    pub fn new(frequency: c_int) -> Self {
+        FlamegraphProfiler {
+            frequency,
+            active_profiler: None,
+        }
+    }
+}
+
+impl<'a> Profiler for FlamegraphProfiler<'a> {
+    fn start_profiling(&mut self, _benchmark_id: &str, _benchmark_dir: &Path) {
+        self.active_profiler = Some(ProfilerGuard::new(self.frequency).unwrap());
+    }
+
+    fn stop_profiling(&mut self, _benchmark_id: &str, benchmark_dir: &Path) {
+        std::fs::create_dir_all(benchmark_dir).unwrap();
+        let flamegraph_path = benchmark_dir.join("flamegraph.svg");
+        let flamegraph_file = File::create(&flamegraph_path).unwrap();
+        self.active_profiler
+            .take()
+            .unwrap()
+            .report()
+            .build()
+            .unwrap()
+            .flamegraph(flamegraph_file)
+            .unwrap();
```

Comment:
Just to be sure, these `unwrap`s won't cause errors? If they do, it would be nice to have a relevant panic message (for example, if you can't create the directory or file)

Otherwise, LGTM

Reply:
They would but i think it is only the case when you run the benchmarks in profiling mode (the command in the description)

Reply:
fixed